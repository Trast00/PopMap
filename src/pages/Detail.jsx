import React from 'react'
import { useSelector } from 'react-redux'
import { useParams } from 'react-router-dom'
import DetailCountry from '../components/DetailCountry/DetailCountry'
import Header from '../components/Header'

function Detail(props) {
  const {topLevelDomain} = useParams()

  const listCountry = useSelector(state => state.countryReducer.listCountries)
  let name = topLevelDomain
  let detailContent = ''
  if(listCountry){
    const country = listCountry.filter(country=> (
      country.numericCode.toString() === topLevelDomain
    ))[0]
    name = country.name
    detailContent = (<DetailCountry country={country}/>)
  }
  return (
    <div>
      <Header name={name} canBack={true} />
      {listCountry && (detailContent)}
    </div>
  )
}

export default Detail
