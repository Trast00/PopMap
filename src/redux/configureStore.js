import { configureStore } from '@reduxjs/toolkit';
import countryReducer from './country/countryReducer';

export default configureStore({
  reducer: {
    countryReducer,
  },
});
