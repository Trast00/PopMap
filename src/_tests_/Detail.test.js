import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from '../redux/configureStore.js';
import Detail from '../pages/Detail.jsx';
import { MemoryRouter } from 'react-router-dom';

describe('test render', () => {
  test('List Rocket should match snapshoot', () => {
    const tree = renderer.create(
      <React.StrictMode>
        <Provider store={configureStore}>
          <MemoryRouter initialEntries={['/test1']}>
            <Detail />
          </MemoryRouter>
        </Provider>
      </React.StrictMode>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
})