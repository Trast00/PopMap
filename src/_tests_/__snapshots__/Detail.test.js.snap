// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render List Rocket should match snapshoot 1`] = `
<div>
  <header
    className="flex-center"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        className="iconBack"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M328 112L184 256l144 144"
          fill="none"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="48"
        />
      </svg>
    </a>
    <p>
      Home
    </p>
    <div />
  </header>
</div>
`;
